src/renderer/kube-object.store.ts

```diff
@@ -71,42 +77,45 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
     }
   }
 
-  protected async loadItems(allowedNamespaces?: string[]): Promise<T[]> {
-    if (!this.api.isNamespaced || !allowedNamespaces) {
-      return this.api.list({}, this.query);
-    } else {
-      return Promise
-        .all(allowedNamespaces.map(namespace => this.api.list({ namespace })))
-        .then(items => items.flat());
+  protected async loadItems({ isAdmin, namespaces, api }: KubeObjectStoreLoadingParams): Promise<T[]> {
+    if (!api.isNamespaced) {
+      if (isAdmin) return api.list({}, this.query);
+
+      return [];
     }
+
+    return Promise
+      .all(namespaces.map(namespace => api.list({ namespace })))
+      .then(items => items.flat());
   }
 
   protected filterItemsOnLoad(items: T[]) {
     return items;
   }
 
   @action
-  async loadAll() {
+  async loadAll(params: { namespaces?: string[] } = {}) {
     this.isLoading = true;
     let items: T[];
 
     try {
-      const { allowedNamespaces, accessibleNamespaces, isAdmin } = getHostedCluster();
+      const { namespaceStore } = await import("./components/+namespaces/namespace.store");
+      const contextNamespaces = params.namespaces || namespaceStore.getContextNamespaces();
```

Comment:
I think this is a bad idea. It implies that there is a circular dependency here. Probably because the namespace store is within the `+namespaces` component folder. I think we should move that out to remove this circular dependency.

Reply:
This is trade-off and there is no circular dependency when loaded via dynamic `import`..
Ideally we should extract logic from namespace-store related to context namespaces, but it means to break current api or create code-duplication.

Reply:
Okay, sounds good. Maybe add a `TODO` comment about refactoring that out.